compressors/arithmetic_coding.py

```diff
@@ -244,9 +177,15 @@ class ArithmeticDecoder(DataDecoder):
     - Charles Bloom's blog: https://www.cbloom.com/algs/statisti.html#A5
     """
 
-    def __init__(self, params: AECParams, freq_base: Frequencies, freq_model_cls: FreqModelBase):
+    def __init__(
+        self, params: AECParams, freq_model_params: Frequencies, freq_model_cls: FreqModelBase
+    ):
         self.params = params
-        self.freq_model = freq_model_cls(freq_base, params.MAX_ALLOWED_TOTAL_FREQ)
+        if isinstance(freq_model_params, tuple):
```

Comment:
created an issue to simplify this.. https://github.com/kedartatwawadi/stanford_compression_library/issues/70
I will look into this later

Reply:
makes sense, I am also not happy with the current hack